File: 0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        # using sliding window technique
        current_cheapest = prices[0]
        current_max_profit = 0
        for i in range(len(prices)):
            if prices[i] < current_cheapest:
                current_cheapest = prices[i]
            if prices[i] - current_cheapest > current_max_profit:
                current_max_profit = prices[i] - current_cheapest
        return current_max_profit

File: 0121-best-time-to-buy-and-sell-stock/solution.c
int maxProfit(int* prices, int pricesSize){
    int maxProfit = 0, localMin = prices[0];
    for (int i = 1; i < pricesSize; i++) {
        if (prices[i] - localMin > maxProfit)
            maxProfit = prices[i] - localMin;
        if (prices[i] < localMin)
            localMin = prices[i];
    }
    return maxProfit;
}
